evmrpc/tx.go

```diff
@@ -240,13 +245,15 @@ func getEthTxForTxBz(tx tmtypes.Tx, decoder sdk.TxDecoder) *ethtypes.Transaction
 
 // Gets the EVM tx index based on the tx index (typically from receipt.TransactionIndex
 // Essentially loops through and calculates the index if we ignore cosmos txs
-func GetEvmTxIndex(txs tmtypes.Txs, txIndex uint32, decoder sdk.TxDecoder) (index int, found bool) {
+func GetEvmTxIndex(txs tmtypes.Txs, txIndex uint32, decoder sdk.TxDecoder, receiptChecker func(common.Hash) bool) (index int, found bool) {
```

Comment:
It's existing code, but I feel like this might be cleaner here
```
func GetEvmTxIndex(txs tmtypes.Txs, txIndex uint32, decoder sdk.TxDecoder, receiptChecker func(common.Hash) bool) (index int, found bool) {
	var evmTxIndex int
	for i, tx := range txs {
		etx := getEthTxForTxBz(tx, decoder)
		
		// does not exist and has no receipt (cosmos)
		if etx == nil && !receiptChecker(sha256.Sum256(tx)) {
			continue
		}
		
		// found the index
		if i == int(txIndex) {
			return evmTxIndex, true
		}
		
		evmTxIndex++
	}
	return -1, false
}
```

Reply:
Just confirming, we DO want to increment the evmTxIndex on a fake receipt, right?

Reply:
yeah we do